code/modules/projectiles/projectile/magic.dm

```diff
@@ -351,21 +351,9 @@
 
 /obj/item/projectile/magic/aoe/Range()
 	if(proxdet)
-		var/turf/T1 = get_step(src,turn(dir, -45))
-		var/turf/T2 = get_step(src,turn(dir, 45))
-		var/turf/T3 = get_step(src,dir)
-		var/mob/living/L = locate(/mob/living) in T1 //if there's a mob alive in our front right diagonal, we hit it.
-		if(L && L.stat != DEAD)
-			Bump(L,1) //Magic Bullet #teachthecontroversy
-			return
-		L = locate(/mob/living) in T2
-		if(L && L.stat != DEAD)
-			Bump(L,1)
-			return
-		L = locate(/mob/living) in T3
-		if(L && L.stat != DEAD)
-			Bump(L,1)
-			return
+		for(var/mob/living/L in range(1, get_turf(src)))
+			if(L.stat != DEAD && L != firer)
+				Bump(L, TRUE)
```

Comment:
This should return on Bump if it is meant to preserve old function of hitting only one mob